### ISSUE TITLE
Should Conv2D's use_bias be set to True?

### ISSUE THREAD

**Issue by jcao-ai** [shaoanlu/MUNIT-keras#3]
I find that `Conv2D` before `InstanceNormalization` set `use_bias` to `True`. Should we just set it to `False` because `InstanceNormalization` includes some kind of `bias`

**Comment by shaoanlu** [comment]
I followed the pytorch implement,  which set `use_bias=True` for all `Conv2dBlock()`, as shown [here](https://github.com/NVlabs/MUNIT/blob/master/networks.py#L257). 

Your point makes sense. The bias of conv2d seems to be shifted back (but not totally zero out) after IN since IN normalizes tensors channel-wise. Can you elaborate more on the relationship between bias and instance normalization? It sound interesting to me because I've seem some GAN implementation using biases and some not. And I don't really know the reason behind it.

**Comment by jcao-ai** [comment]
@shaoanlu 
> From **instance_normalization.py**

```python
def call(self, inputs, training=None):
        input_shape = K.int_shape(inputs)
        reduction_axes = list(range(0, len(input_shape)))

        if (self.axis is not None):
            del reduction_axes[self.axis]

        del reduction_axes[0]

        mean = K.mean(inputs, reduction_axes, keepdims=True)
        stddev = K.std(inputs, reduction_axes, keepdims=True) + self.epsilon
        normed = (inputs - mean) / stddev

        broadcast_shape = [1] * len(input_shape)
        if self.axis is not None:
            broadcast_shape[self.axis] = input_shape[self.axis]

        if self.scale:
            broadcast_gamma = K.reshape(self.gamma, broadcast_shape)
            normed = normed * broadcast_gamma
        if self.center:
            broadcast_beta = K.reshape(self.beta, broadcast_shape)
            normed = normed + broadcast_beta
        return normed
```

Let's suppose the `inputs = X+bias`, then this InstanceNormalization is something like:
```
output = normalized(X+bias)*gamma + beta
```
Actually `normalized` step will ignore `bias` term because `normed = (inputs - mean) / stddev` where the values of molecular and denominator are unchanging.

In addition,  there is a `beta` term here which act as a `bias` role.

**Comment by shaoanlu** [comment]
Thanks! I understand it now. Normalization is a linear operation, adding bias to its input doesn't affect the normalized output.

Although being trivial, I ran a toy experiment on fashion MNIST:
```python
LARGE_BIAS = 1e5

model = Sequential()
model.add(Flatten(input_shape=input_shape))
model.add(Dense(64, bias_initializer=keras.initializers.Constant(value=LARGE_BIAS)))
model.add(InstanceNormalization())
model.add(Activation('relu'))
model.add(Dense(32, bias_initializer=keras.initializers.Constant(value=LARGE_BIAS)))
model.add(InstanceNormalization())
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(num_classes, activation='softmax'))
```
The validation accuracy was 87% w/ IN and 10% w/o IN after 10 epochs. The same hold true for batch/layer/group norm.

**Edit:** In tensorflow, bias variables will not be created if `normalizer_fn` is provided ([souce](https://github.com/tensorflow/tensorflow/issues/14173)). So there are some `use_bias=True`  in my implementation being redundant.

**Comment by jcao-ai** [comment]
@shaoanlu You are right. And I am glad to help others who are confused with this problem. 

Maybe a pull request is needed soon.
